src/compiler/checker.ts

```diff
@@ -6925,7 +6966,11 @@ namespace ts {
             checkCollisionWithCapturedThisVariable(node, node);
             checkBlockScopedBindingCapturedInLoop(node, symbol);
 
-            return getNarrowedTypeOfSymbol(getExportSymbolOfValueSymbolIfExported(symbol), node);
+            const type = getNarrowedTypeOfSymbol(getExportSymbolOfValueSymbolIfExported(symbol), node);
+            if (type === undefinedType || type == nullType) {

```

Comment:
This checks for the undefined type instead of the symbol with the name "undefined". I think this is wrong, so I changed it to `if (symbol.name === "undefined")`